File: ShoppingList2000Backend/Domain/Entities/ShoppingList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Entities
{
    public class ShoppingList
    {
        public string ShoppingListId { get; set; }
        public string ShoppingListName { get; set; }
        public List<Product> Products { get; set; }
        public string CreatorUserId { get; set; }
        public List<string> EligebleUsers { get; set; }

    }
}

File: ShoppingList2000Frontend/src/models/ShoppingList.ts
import { Product } from './Product';

export class ShoppingList {
  shoppingListId: string | undefined | null;
  shoppingListName: string;
  products: Product[];
  creatorUserId: string;
  eligibleUsers: string[];
  lastUpdatedUser: string | undefined | null;

  constructor(shoppingListName: string, creatorUserId: string, eligibleUsers: string[], products: Product[] = [], lastUpdatedUser: string) {
    this.shoppingListId = null;
    this.shoppingListName = shoppingListName;
    this.creatorUserId = creatorUserId;
    this.eligibleUsers = eligibleUsers;
    this.products = products.map(product => new Product(product.name, product.isPermanent, product.isChecked));
    this.lastUpdatedUser = lastUpdatedUser;
  }
}